peer/peerstorage_test.go

```diff
@@ -0,0 +1,69 @@
+package peer
+
+import (
+	"os"
+	"testing"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/lightningnetwork/lnd/kvdb"
+	"github.com/stretchr/testify/require"
+)
+
+// TestKvdbPeerStorage tests the KvdbPeerStorage functionality.
+func TestKvdbPeerStorage(t *testing.T) {
+	rt := require.New(t)
+
+	file, err := os.CreateTemp("", "*.db")
+	rt.NoError(err, "unable to create temp file")
+	t.Cleanup(func() {
+		rt.NoError(file.Close())
+		rt.NoError(os.Remove(file.Name()))
+	})
+
+	dbPath := file.Name()
+	db, err := kvdb.Open(
+		kvdb.BoltBackendName, dbPath, true, kvdb.DefaultDBTimeout,
+	)
+	t.Cleanup(func() {
+		rt.NoError(db.Close())
+	})
+
+	rt.NoError(err, "unexpected err creating kvdb DB")
+
+	// Create a KvdbPeerStorage instance.
+	peerDataStore, err := NewKvdbPeerStorage(db)
+	rt.NotNil(peerDataStore)
+	rt.NoError(err, "unexpected error creating new "+
+		"KvdbPeerStorage instance.")
+
+	// Create a sample private key for testing.
+	samplePrivKey, _ := btcec.NewPrivateKey()
+
+	// Test StorePeerData.
+	err = peerDataStore.Store([]byte("sample data"),
+		*samplePrivKey.PubKey())
+	rt.NoError(err, "unexpected error storing peer "+
+		"data.")
+
+	// Test RetrievePeerData.
+	retrievedData, err := peerDataStore.Retrieve(*samplePrivKey.
+		PubKey())
+
+	rt.NoError(err, "unexpected error retrieving peer "+
+		"data.")
+
+	rt.NotNil(retrievedData)
+
+	// Test DeletePeerData.
+	err = peerDataStore.Delete(*samplePrivKey.PubKey())
+	rt.NoError(err, "unexpected error deleting peer data.")
+
+	// Test that there is no data stored for that peer.
+	retrievedData, err = peerDataStore.Retrieve(*samplePrivKey.
```

Comment:
you should that the retrievedData is equal to `[]byte("sample data")`

Reply:
To clarify, (I think) Elle is saying that this should be checked earlier instead of simply asserting it is non-nil